src/test/java/com/neo4j/docker/TestConfSettings.java

```diff
@@ -135,4 +136,53 @@ void testReadTheConfFile() throws Exception
         //Kill the container
         container.stop();
     }
+
+    @Test
+    void testCommentedConfigsAreReplacedByDefaultOnes() throws Exception
+    {
+        //Create container
+        GenericContainer container = createContainer();
+        //Mount /conf
+        Path confMount = HostFileSystemOperations.createTempFolderAndMountAsVolume( container, "conf-", "/conf" );
+        SetContainerUser.currentlyRunningUser( container );
+        //Create neo4j.conf file
+        Path confFile = Paths.get( "src", "test", "resources", "neo4j.conf" );
+        Files.copy(confFile, confMount.resolve("neo4j.conf"));
+        //Start the container
+        container.setWaitStrategy( Wait.forLogMessage( ".*Config Dumped*." , 1 ).withStartupTimeout( Duration.ofSeconds( 10 ) ) );
+        container.setCommand("dump-config");
+        container.start();
+        //Read the config file to check if the config is appended correctly
+        Stream<String> lines = Files.lines(confMount.resolve("neo4j.conf"));
+        Optional<String> pagecacheSizeMatch = lines.filter(s -> s.contains("dbms.memory.pagecache.size=512M")).findFirst();
+        lines.close();
+        Assertions.assertTrue(pagecacheSizeMatch.isPresent(),"conf settings not appended correctly by docker-entrypoint");
+        //Kill the container
+        container.stop();
+    }
+
+
+    @Test
+    void testConfigsAreNotOverridenByDockerentrypoint() throws Exception
+    {
+        //Create container
+        GenericContainer container = createContainer();
+        //Mount /conf
+        Path confMount = HostFileSystemOperations.createTempFolderAndMountAsVolume( container, "conf-", "/conf" );
+        SetContainerUser.currentlyRunningUser( container );
+        //Create neo4j.conf file
+        Path confFile = Paths.get( "src", "test", "resources", "neo4j.conf" );
+        Files.copy(confFile, confMount.resolve("neo4j.conf"));
+        //Start the container
+        container.setWaitStrategy( Wait.forHttp( "/" ).forPort( 7474 ).forStatusCode( 200 ) );
+        container.start();
+        //Read the config file to check if the config is not overriden
+        Stream<String> lines = Files.lines(confMount.resolve("neo4j.conf"));
```

Comment:
This test is copy a test config file to the location at `confMount.resolve("neo4j.conf")` then starting neo4j.
Since we're starting neo4j in this test rather than doing `dump-config`, that file is never overridden. So what we're actually testing here is that the test conf we copied over has a configuration line that we know we set. Does that make sense?

There are 2 solutions:
1) do a `dump-config` command instead of starting neo4j like `testCommentedConfigsAreReplacedByDefaultOnes()` does.
2) check the log output from neo4j like the test `testReadTheConfFile()` does.

(I think option 1 is better. Since we don't really need to be starting neo4j, it'll be much quicker).